oi/type_graph/RemoveMembers.cpp

```diff
@@ -54,19 +54,12 @@ void RemoveIgnored::visit(Class& c) {
     accept(child);
   }
 
-  for (size_t i = 0; i < c.members.size(); i++) {
-    if (!ignoreMember(c.name(), c.members[i].name)) {
-      continue;
-    }
-    auto& primitive = typeGraph_.makeType<Primitive>(Primitive::Kind::Int8);
-    auto& paddingArray =
-        typeGraph_.makeType<Array>(primitive, c.members[i].type().size());
-    c.members[i] =
-        Member{paddingArray, AddPadding::MemberPrefix, c.members[i].bitOffset};
-  }
+  std::erase_if(c.members, [this, &c](Member member) {
+    return ignoreMember(c.name(), member.name);
+  });
 }
 
-bool RemoveIgnored::ignoreMember(const std::string& typeName,
+bool RemoveMembers::ignoreMember(const std::string& typeName,
```

Comment:
Unrelated but nearby, why is this not checking a set? It seems better to implement a hash function for `std::pair<std::string, std::string>` than loop over a vector of pairs. Even a map of ContainerName -> Set<FieldName> would be a lot better.

Reply:
The real reason: It's sharing the implementation with CodeGen v1 and I didn't want to change too much.

But in terms of efficiency, I think the winner of vectors vs maps/sets would come down to how many elements are typically being stored. For a small number of elements a vector would likely win due to the lack of hashing and being more cache friendly.